src/fabric_cicd/publish.py

```diff
@@ -63,71 +76,111 @@ def publish_all_items(fabric_workspace_obj: FabricWorkspace, item_name_exclude_r
         )
         fabric_workspace_obj.publish_item_name_exclude_regex = item_name_exclude_regex
 
+    # Collect published items information
+    published_items = {}
+    
+    # Helper function to add items to the published_items dictionary
+    def collect_published_items_for_type(item_type: str) -> None:
+        if item_type in fabric_workspace_obj.repository_items:
+            published_items[item_type] = {}
+            for item_name, item_obj in fabric_workspace_obj.repository_items[item_type].items():
```

Comment:
Add differentiation between what was published in this deployment and what already exists in the deployed workspace